### PR TITLE
Remove V2JobStatusMonitor and use LifecycleAgentStatusMonitor as the default instead

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

Further V2 engine cleanup. `V2JobStatusMonitor` was the default `AgentStatusMonitor` in OSS, and
had to be replaced with something else. `LifecycleAgentStatusMonitor` is now the default agent 
health provider.

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    private Optional<AgentStatus> findStatus(String agentInstanceId) {
+        return agentManagementService.findAgentInstance(agentInstanceId).map(this::statusOf);
+    }
+
+    private AgentStatus statusOf(AgentInstance agent) {
+        InstanceLifecycleState state = agent.getLifecycleStatus().getState();
+        if (state == InstanceLifecycleState.Started) {
+            return AgentStatus.healthy(SOURCE_ID, agent, "Agent in Started state", System.currentTimeMillis());
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java]: use TItusRuntime clock?

File: titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        return AgentStatus.unhealthy(SOURCE_ID, agent, "Agent not in Started state: currentState=" + state, System.currentTimeMillis());
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java (hunk 1 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java]: Fixed

File: titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java (hunk 2 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/agent/service/monitor/LifecycleAgentStatusMonitor.java]: Fixed

**Pull request merged**